.github/workflows/tf-deploy-dflook.yaml

```diff
@@ -0,0 +1,127 @@
+on:
+  workflow_call:
+    inputs:
+      aws_account_id:
+        description: "The AWS account ID"
+        type: string
+        required: true
+      aws_region:
+        description: "The AWS region"
+        required: true
+        type: string
+      aws_role_name:
+        description: "The name of the role to assume with OIDC"
+        type: string
+        required: false
+        default: cicd-iac
+      environment:
+        description: "The environment name"
+        type: string
+        required: true
+      github_feedback:
+        description: "Whether to post a comment on the PR with the Terraform plan"
+        required: false
+        type: string
+        default: "false"
+      tf_dir:
+        description: "The directory where the Terraform files are located"
+        type: string
+        required: false
+        default: '.'
+      tf_backend_config:
+        description: "Terraform backend configuration"
+        required: false
+        type: string
+      tf_var_files:
+        description: "Comma separated list of Terraform variable files to pass to the plan command"
+        required: false
+        type: string
+        default: ''
+      tf_vars:
+        description: "Comma separated list of Terraform variables to pass to the plan command"
+        required: false
+        type: string
+        default: ''
+
+jobs:
+  terraform:
+    permissions:
+      contents: read
+      id-token: write
+      pull-requests: write
+    concurrency:
+      group: ${{ github.workflow }}-${{ inputs.environment }}
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: arn:aws:iam::${{ inputs.aws_account_id }}:role/${{ inputs.aws_role_name }}
+          aws-region: ${{ inputs.aws_region }}
+
+      - name: Terraform Format
+        uses: dflook/terraform-fmt-check@v1
+        with:
+          path: ${{ inputs.tf_dir }}
+          backend_config: ${{ inputs.tf_backend_config }}
+
+      - name: Terraform Validate
+        uses: dflook/terraform-validate@v1
+        with:
+          path: ${{ inputs.tf_dir }}
+          backend_config: ${{ inputs.tf_backend_config }}
+
+      - name: Terraform Plan
+        if: github.event_name == 'pull_request'
```

Comment:
Is this condition really needed ? I wanted to test the tf-deploy-dflook-yaml workflow by triggering the caller workflow manually and it doesn't work :-(